src/JakobSteinn/HTMLFormValidator/Validator.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace JakobSteinn\HTMLFormValidator;
+
+use DiDom\Query;
+use DiDom\Document;
+use Zend\InputFilter\Input;
+use Zend\InputFilter\InputFilter;
+use Zend\Validator\ValidatorPluginManager;
+
+class Validator
+{
+    /**
+     * HTML document that should contain a form
+     *
+     * @var DiDom\Document
+     */
+    private $DOMDocument;
+
+    /**
+     * Validation rules for the given form
+     * 
+     * @var array
+     */
+    private $rules = [];
+
+    /**
+     * The selector used to find user form
+     * 
+     * @var string
+     */
+    private $formSelector = '//form[1]/input';
+
+    /**
+     * Instantiate a new Validator instance
+     * 
+     * @param string $html 		The HTML to search through, to find a form
+     */
+    public function __construct($html, $uniqueFormSelector = '')
+    {
+        $this->DOMDocument = new Document($html);
+
+        $this->buildFormSelector($uniqueFormSelector);
+        $this->parseForRules();
+    }
+
+    /**
+     * Validate given data with form rules
+     * 
+     * @param  array  $formData The given form data
+     * @return Zend\InputFilter\InputFilter
+     */
+    public function validate(array $formData)
+    {
+        $inputFilter = new InputFilter();
+
+        foreach ($this->rules as $field => $rule) {
+            $currentInput = new Input($field);
+            $currentInput->getValidatorChain()
+                ->attachByName($rule);
+
+            $inputFilter->add($currentInput);
+        }
+
+        $inputFilter->setData($formData);
+
+        return $inputFilter;
+    }
+
+    /**
+     * Get the document under validation
+     * 
+     * @return DiDom\Document
+     */
+    public function getDocument()
+    {
+        return $this->DOMDocument;
+    }
```

Comment:
Any reason to expose `getDocument()` and `getRules()` as public? It seems like it's breaking encapsulation by exposing internal stuff.